sky/execution.py

```diff
@@ -217,21 +217,21 @@ def _execute(
             if idle_minutes_to_autostop is not None:
                 backend.set_autostop(handle,
                                      idle_minutes_to_autostop,
-                                     terminate=terminate)
+                                     down=down)
 
         if stages is None or Stage.EXEC in stages:
             try:
                 global_user_state.update_last_use(handle.get_cluster_name())
                 backend.execute(handle, task, detach_run)
             finally:
                 # Enables post_execute() to be run after KeyboardInterrupt.
-                backend.post_execute(handle, terminate)
-
-        if stages is None or Stage.TERMINATE in stages:
-            if terminate and idle_minutes_to_autostop is None:
+                backend.post_execute(handle, down)
+    finally:
+        if stages is None or Stage.DOWN in stages:
+            if down and idle_minutes_to_autostop is None:
                 backend.teardown_ephemeral_storage(task)
                 backend.teardown(handle, terminate=True)
```

Comment:
> Yes, it will clean up after the job fails, but if the cluster fails at the setup/filemount.
During adding this I found that it previously will not terminate the cluster when the user ctrl-c'ed during the logging for the execution. I just moved the teardown into the finally: ... block to make sure the clean up will happen after the ctrl-c. Wdyt?

Do you mean that by putting this block in this `finally`, we will tear down when users ctrl-c a running job's streaming log? That seems unintuitive. More intuitive thing to do seems to be terminate it whenever the job finishes.

By "but if the cluster fails at the setup/filemount." - do you mean if these happen, the cluster will remain up?  This is important to clarify in all func docstrs/help strs.

Reply:
Oh, my bad, I meant when the `--down` or `sky.launch(..., down=True, ...)` is specified, the cluster should be terminated if the user press ctrl-c during setup or file mounts. After thinking more about this, I think it should be fine to place it back in its original place.